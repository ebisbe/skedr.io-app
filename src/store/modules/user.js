import Auth from '@aws-amplify/auth'
import LogRocket from 'logrocket'

export const state = {
  user: null,
  username: null,
  userId: null,
  userVerification: null
}

export const getters = {
  userId: state => state.username
}

export const mutations = {
  setUser(state, user) {
    state.user = user
    if (user !== null) {
      state.username = decodeURIComponent(user.username)
    } else {
      state.username = null
    }
  },
  setUserAttributes(state, attributes) {
    state.user.attributes = Object.assign({}, state.user.attributes, attributes)
  },
  setUserId: (state, userId) => (state.userId = userId),
  setUserVerification: (state, data) => {
    state.userVerification = data
  }
}

export const actions = {
  loginUser: async function({ state, commit }, { username, password }) {
    // the value of `username` is really an `email`
    const user = await Auth.signIn(username.toLowerCase(), password)
    commit('setUser', user)

    const data = await Auth.verifiedContact(user)
    commit('setUserVerification', data)

    const payload = {
      body: { userId: state.username, email: username }
    }
    return Object.keys(data.verified).length !== 0
  },
  currentAuthenticatedUser: async ({ commit, state }) => {
    try {
      const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
      const credentials = await Auth.currentCredentials()
      commit('setUser', user)
      commit('setUserId', credentials.identityId)
      LogRocket.identify(state.username, {
        name: `${user.attributes.name} ${user.attributes.family_name}`,
        email: user.attributes.email
      })
      LogRocket.getSessionURL(sessionURL => {
        const indentifyUser = {
          name: `${user.attributes.name} ${user.attributes.family_name}`,
          email: user.attributes.email,
          sessionURL: sessionURL
        }
        Beacon('identify', indentifyUser)
      })
    } catch (err) {
      commit('setUser', null)
      return false
    }
    return true
  },
  updateUserAttributes: async ({ commit, state }) => {
    await Auth.updateUserAttributes(state.user, state.user.attributes)
    const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
    commit('setUser', user)
  },
  confirmEmailCode: async ({ commit }, code) => {
    await Auth.verifyCurrentUserAttributeSubmit('email', code)
    const user = await Auth.currentAuthenticatedUser({ bypassCache: true })
    commit('setUser', user)
  },
  signOut: async ({ state }) => {
    if (state.user.attributes.email_verified === false) {
      throw {
        name: 'VerifyEmailError',
        message: 'To sign out first verify your email'
      }
    }
    await Auth.signOut()
    return true
  }
}
