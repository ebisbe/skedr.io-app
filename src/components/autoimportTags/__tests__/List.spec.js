import { shallowMount } from '@vue/test-utils'
import Component from '../List.vue'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import { getters } from '@/store/modules/tagsFilter'
Vue.use(Vuetify)
Vue.use(Vuex)

const createCmp = ({ propsData, mocks, store }) =>
  shallowMount(Component, {
    propsData,
    mocks,
    store
  })
const createStore = items =>
  new Vuex.Store({
    modules: {
      tagsFilter: {
        namespaced: true,
        state: {
          items
        },
        getters
      }
    }
  })
describe('List.vue', () => {
  it('is loading', () => {
    const mocks = {
      $apollo: {
        queries: {
          groupTagsList: {
            loading: true
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    expect(wrapper).toMatchSnapshot()
  })
  it('has an error', () => {
    const mocks = {
      $apollo: {
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    wrapper.setData({ error: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('has no results', () => {
    const mocks = {
      $apollo: {
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const wrapper = createCmp({ mocks })
    wrapper.setData({
      groupTagsList: []
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('has 3 results one is filtered', () => {
    const mocks = {
      $apollo: {
        error: false,
        queries: {
          groupTagsList: {
            loading: false
          }
        }
      }
    }
    const store = createStore({ tag1: false, tag2: true })
    const wrapper = createCmp({ mocks, store })
    wrapper.setData({
      groupTagsList: [
        { group: { title: 'B' }, tags: ['tag2', 'tagB'], groupId: 'groupId1' },
        { group: { title: 'A' }, tags: ['tag2', 'tagA'], groupId: 'groupId2' },
        { group: { title: ':a' }, tags: ['taga'], groupId: 'groupId3' }
      ]
    })
    expect(wrapper).toMatchSnapshot()
  })
})
