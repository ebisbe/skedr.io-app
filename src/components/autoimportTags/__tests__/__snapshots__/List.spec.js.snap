// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue has 3 results one is filtered 1`] = `
<v-content-stub tag="main">
  <payment-banner-stub></payment-banner-stub>
  <v-container-stub tag="div">
    <ol>
      <li>
        You don't have any tag linked to a group. To start linking tags search through your groups
        or find new groups with the
        <v-icon-stub>search</v-icon-stub>Search.
      </li>
      <li>
        Add tags with the <v-icon-stub>settings</v-icon-stub> icon of any group. For example:
        <ul>
          <li><a href="https://www.flickr.com/groups/flickr10photowalks/">Flickr Worldwide Photowalks</a>
            use the <group-tag-dialog-chip-stub tag="photowalk"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/peopleandpersons/">People and Persons</a>
            use the <group-tag-dialog-chip-stub tag="people"></group-tag-dialog-chip-stub>,
            <group-tag-dialog-chip-stub tag="person"></group-tag-dialog-chip-stub>, <group-tag-dialog-chip-stub tag="woman"></group-tag-dialog-chip-stub> and/or
            <group-tag-dialog-chip-stub tag="man"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/catchy/">!Catchy Colors</a>
            use the <group-tag-dialog-chip-stub tag="color"></group-tag-dialog-chip-stub> or
            <group-tag-dialog-chip-stub tag="colour"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/minimally_less_is_more/">Less Is More...</a>
            use the <group-tag-dialog-chip-stub tag="minimalism"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/blackandwhite/">Black and White</a>
            use the <group-tag-dialog-chip-stub tag="b&amp;w"></group-tag-dialog-chip-stub> tag.
          </li>
        </ul>
      </li>
      <li>
        Once the tag/s has been linked to the group all your previous photos with that tag will be
        shared to that group automatically. New photos you upload will be added too.
      </li>
    </ol>
    <v-flex-stub tag="div" xs12="" class="pt-12">
      <h3>How it works?</h3> <video src="/video/autoimport/Video MP4 (1180x808).mp4" width="640" controls="controls"></video>
    </v-flex-stub>
  </v-container-stub>
</v-content-stub>
`;

exports[`List.vue has an error 1`] = `
<v-content-stub tag="main">
  <payment-banner-stub></payment-banner-stub>
  <v-container-stub tag="div">
    <ol>
      <li>
        You don't have any tag linked to a group. To start linking tags search through your groups
        or find new groups with the
        <v-icon-stub>search</v-icon-stub>Search.
      </li>
      <li>
        Add tags with the <v-icon-stub>settings</v-icon-stub> icon of any group. For example:
        <ul>
          <li><a href="https://www.flickr.com/groups/flickr10photowalks/">Flickr Worldwide Photowalks</a>
            use the <group-tag-dialog-chip-stub tag="photowalk"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/peopleandpersons/">People and Persons</a>
            use the <group-tag-dialog-chip-stub tag="people"></group-tag-dialog-chip-stub>,
            <group-tag-dialog-chip-stub tag="person"></group-tag-dialog-chip-stub>, <group-tag-dialog-chip-stub tag="woman"></group-tag-dialog-chip-stub> and/or
            <group-tag-dialog-chip-stub tag="man"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/catchy/">!Catchy Colors</a>
            use the <group-tag-dialog-chip-stub tag="color"></group-tag-dialog-chip-stub> or
            <group-tag-dialog-chip-stub tag="colour"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/minimally_less_is_more/">Less Is More...</a>
            use the <group-tag-dialog-chip-stub tag="minimalism"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/blackandwhite/">Black and White</a>
            use the <group-tag-dialog-chip-stub tag="b&amp;w"></group-tag-dialog-chip-stub> tag.
          </li>
        </ul>
      </li>
      <li>
        Once the tag/s has been linked to the group all your previous photos with that tag will be
        shared to that group automatically. New photos you upload will be added too.
      </li>
    </ol>
    <v-flex-stub tag="div" xs12="" class="pt-12">
      <h3>How it works?</h3> <video src="/video/autoimport/Video MP4 (1180x808).mp4" width="640" controls="controls"></video>
    </v-flex-stub>
  </v-container-stub>
</v-content-stub>
`;

exports[`List.vue has no results 1`] = `
<v-content-stub tag="main">
  <payment-banner-stub></payment-banner-stub>
  <v-container-stub tag="div">
    <ol>
      <li>
        You don't have any tag linked to a group. To start linking tags search through your groups
        or find new groups with the
        <v-icon-stub>search</v-icon-stub>Search.
      </li>
      <li>
        Add tags with the <v-icon-stub>settings</v-icon-stub> icon of any group. For example:
        <ul>
          <li><a href="https://www.flickr.com/groups/flickr10photowalks/">Flickr Worldwide Photowalks</a>
            use the <group-tag-dialog-chip-stub tag="photowalk"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/peopleandpersons/">People and Persons</a>
            use the <group-tag-dialog-chip-stub tag="people"></group-tag-dialog-chip-stub>,
            <group-tag-dialog-chip-stub tag="person"></group-tag-dialog-chip-stub>, <group-tag-dialog-chip-stub tag="woman"></group-tag-dialog-chip-stub> and/or
            <group-tag-dialog-chip-stub tag="man"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/catchy/">!Catchy Colors</a>
            use the <group-tag-dialog-chip-stub tag="color"></group-tag-dialog-chip-stub> or
            <group-tag-dialog-chip-stub tag="colour"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/minimally_less_is_more/">Less Is More...</a>
            use the <group-tag-dialog-chip-stub tag="minimalism"></group-tag-dialog-chip-stub> tag.
          </li>
          <li><a href="https://www.flickr.com/groups/blackandwhite/">Black and White</a>
            use the <group-tag-dialog-chip-stub tag="b&amp;w"></group-tag-dialog-chip-stub> tag.
          </li>
        </ul>
      </li>
      <li>
        Once the tag/s has been linked to the group all your previous photos with that tag will be
        shared to that group automatically. New photos you upload will be added too.
      </li>
    </ol>
    <v-flex-stub tag="div" xs12="" class="pt-12">
      <h3>How it works?</h3> <video src="/video/autoimport/Video MP4 (1180x808).mp4" width="640" controls="controls"></video>
    </v-flex-stub>
  </v-container-stub>
</v-content-stub>
`;

exports[`List.vue is loading 1`] = `
<v-content-stub tag="main">
  <payment-banner-stub></payment-banner-stub>
  <q-empty-stub icon="" description="" loading="true"></q-empty-stub>
</v-content-stub>
`;
